src/main/java/org/ibp/api/rest/germplasm/GermplasmNameResource.java

```diff
@@ -33,6 +38,24 @@
 		return new ResponseEntity<>(this.germplasmNameService.createName(programUUID, germplasmNameRequestDto, gid), HttpStatus.OK);
 	}
 
+	@ApiOperation(value = "Create code name (CODE1, CODE2, CODE2) for specified list of germplasm")
+	@PreAuthorize("hasAnyAuthority('ADMIN', 'GERMPLASM', 'MANAGE_GERMPLASM', 'CODE_GERMPLASM')")
+	@RequestMapping(value = "/crops/{cropName}/germplasm/coding", method = RequestMethod.POST)
+	@ResponseBody
+	public ResponseEntity<List<GermplasmGroupNamingResult>> createGermplasmCodeNames(@PathVariable final String cropName,
+		@RequestBody final GermplasmCodeNameBatchRequestDto germplasmCodeNameBatchRequestDto) {
+		return new ResponseEntity<>(this.germplasmNameService.createCodeNames(germplasmCodeNameBatchRequestDto), HttpStatus.OK);
+	}
+
+	@ApiOperation(value = "Get next code name sequence based on the specified name settings")
+	@PreAuthorize("hasAnyAuthority('ADMIN', 'GERMPLASM', 'MANAGE_GERMPLASM', 'CODE_GERMPLASM')")
+	@RequestMapping(value = "/crops/{cropName}/germplasm/coding/sequence", method = RequestMethod.POST)
```

Comment:
Sequence generation for germplasm naming could also apply to crosses so we should ake the endpoint generic (and also modify the API description to be generic).
What do you think about POST /crops/{cropName}/germplasm/names/next-name-generation?

Reply:
Yeah, I think that's more appropriate.

Reply:
Done